src/hotspot/share/services/diagnosticArgument.cpp

```diff
@@ -195,6 +195,11 @@ template <> void DCmdArgument<char*>::parse_value(const char* str,
   } else {
     // Use realloc as we may have a default set.
     if (strcmp(type(), "FILE") == 0) {
+      if (str == nullptr || *str == 0) {
+        stringStream error_msg;
+        error_msg.print("Filename is empty or not specified. %s", str);
+        THROW_MSG(vmSymbols::java_lang_IllegalArgumentException(), error_msg.base());
+      }
```

Comment:
In case anyone else thinks the same thing: This won't access `error_msg.base()` after returning as `THROW_MSG` copies the message.